osu.Game/Rulesets/Mods/Mod.cs

```diff
@@ -42,6 +47,44 @@ public abstract class Mod : IMod, IJsonSerializable
         [JsonIgnore]
         public virtual string Description => string.Empty;
 
+        /// <summary>
+        /// The tooltip to display for this mod when used in a <see cref="ModIcon"/>.
+        /// </summary>
+        /// <remarks>
+        /// Differs from <see cref="Name"/>, as the value of attributes (AR, CS, etc) changeable via the mod
+        /// are displayed in the tooltip.
+        /// </remarks>
+        [JsonIgnore]
+        public virtual string IconTooltip
+        {
+            get
+            {
+                List<string> attributes = new List<string>();
+                foreach ((SettingSourceAttribute attr, System.Reflection.PropertyInfo property) in this.GetOrderedSettingsSourceProperties())
+                {
+                    // use TooltipText from SettingSource if available, but fall back to Label, which has to be provided
+                    string tooltipText = attr.TooltipText ?? attr.Label + " {0}";
+                    object bindableObj = property.GetValue(this);
+                    if (bindableObj is BindableInt bindableInt && !bindableInt.IsDefault)
```

Comment:
couldn't find a way to cast to a generic bindable, am I missing something? Could cast to `IBindable` but not `IBindable<T>`, the latter of which has the `Value` method.

Reply:
You can't cast to a generic type without knowing the actual type parameters. That said, I don't think casting to `Bindable<T>` is necessary, as [`Bindable<T>`'s default implementation for `ToString()` returns the value](https://github.com/ppy/osu-framework/blob/7b8a966bbddc5fc09aec38bfe9df6e7440409a9a/osu.Framework/Bindables/Bindable.cs#L377). You can check if the value is default without knowing the type param by casting to [`IHasDefaultValue`](https://github.com/ppy/osu-framework/blob/master/osu.Framework/Bindables/IHasDefaultValue.cs) instead.

So in general you'd probably want to do something like

```csharp
var settingIsDefault = (bindableObj as IHasDefaultValue)?.IsDefault;
var settingValueText = settingIsDefault == true ? string.Empty : bindableObj.ToString();
```
Note that the `== true` comparison is not redundant in this case, as the null-coalescing operator makes `settingIsDefault` be a tri-state `bool?`. This get rid of all the ugly type checks.